Userland/Applications/PixelPaint/Tools/PolygonalSelectTool.cpp

```diff
@@ -0,0 +1,244 @@
+                continue;
+            }
+            if (flood_mask.get(candidate_point.x(), candidate_point.y()) == 0 && polygon_bitmap.get_pixel(candidate_point.x(), candidate_point.y()) != Color::Black) {
+
+                polygon_bitmap.set_pixel(candidate_point.x(), candidate_point.y(), Color::Black);
+                points_to_visit.enqueue(candidate_point);
+                selection_mask.set(Gfx::IntPoint(candidate_point.x(), candidate_point.y()).translated(polygon_delta), 0);
+            }
+            flood_mask.set(candidate_point.x(), candidate_point.y(), 0xFF);
+        }
+    }
+
+    selection_mask.shrink_to_fit();
+    selection_mask.bounding_rect().translate_by(m_editor->active_layer()->location());
+    m_editor->image().selection().merge(selection_mask, m_merge_mode);
+}
+
+void PolygonalSelectTool::process_polygon()
+{
+    // Determine minimum bounding box that can hold the polygon.
+    auto min_x_seen = m_polygon_points.at(0).x();
+    auto max_x_seen = m_polygon_points.at(0).x();
+    auto min_y_seen = m_polygon_points.at(0).y();
+    auto max_y_seen = m_polygon_points.at(0).y();
+
+    for (auto point : m_polygon_points) {
+        if (point.x() < min_x_seen)
+            min_x_seen = point.x();
+        if (point.x() > max_x_seen)
+            max_x_seen = point.x();
+        if (point.y() < min_y_seen)
+            min_y_seen = point.y();
+        if (point.y() > max_y_seen)
+            max_y_seen = point.y();
+    }
+
+    // We create a bitmap that is bigger by 1 pixel on each side (+2) and need to account for the 0 indexed
+    // pixel positions (+1) so we make the bitmap size the delta of x/y min/max + 3
+    auto polygon_bitmap_or_error = Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRA8888, { (max_x_seen - min_x_seen) + 3, (max_y_seen - min_y_seen) + 3 });
+    if (polygon_bitmap_or_error.is_error())
+        return;
+
+    auto polygon_bitmap = polygon_bitmap_or_error.release_value();
+
+    auto polygon_painter = Gfx::Painter(polygon_bitmap);
+
+    // We want to paint the polygon into the bitmap such that there is an empty 1px border all the way around it
+    // this ensures that we have a known pixel (0,0) that is outside the polygon. Since the coordinates are relative
+    // to the layer but the bitmap is cropped to the bounding rect of the polygon we need to offset our
+    // points by the the negative of min x/y. And because we want a 1 px offset to the right and down, we + 1 this.
+    auto polygon_bitmap_delta = Gfx::IntPoint(-min_x_seen + 1, -min_y_seen + 1);
+    for (size_t i = 0; i < m_polygon_points.size() - 1; i++) {
+        polygon_painter.draw_line(m_polygon_points.at(i).translated(polygon_bitmap_delta), m_polygon_points.at(i + 1).translated(polygon_bitmap_delta), Color::Black);
+    }
+    polygon_painter.draw_line(m_polygon_points.at(m_polygon_points.size() - 1).translated(polygon_bitmap_delta), m_polygon_points.at(0).translated(polygon_bitmap_delta), Color::Black);
```

Comment:
You said you fixed my previous comment here but I see no change, perhaps you forgot to push it?

Reply:
Hmmm, I'll take a look. Sorry about that, I'll get it fixed properly

Reply:
I just force pushed after rebasing and resolving some other issues. This should be fixed now.